### ISSUE TITLE
PA Computer using invalid construction graph when build from scratch

### ISSUE THREAD

**Issue by tetonis** [space-wizards/space-station-14#18511]
When the PA Computer is built manually, it ends up using an invalid node resulting in it being impossible to complete it. In fact, it is using the wrong construction graph entirely. 

### Built From Scratch:
![image](https://github.com/space-wizards/space-station-14/assets/47284450/ae99a528-349a-4fd6-9248-b74237ac162f)

### Spawned in:
![image](https://github.com/space-wizards/space-station-14/assets/47284450/09ce4551-e723-4892-8e6a-002871df9725)

So obviously this is not supposed to happen and makes it impossible to set up a PA without one being already in the station. This ruins the fun of building your own generator from scratch.

### Also:
https://github.com/space-wizards/space-station-14/assets/47284450/72123647-718f-45fe-8b41-4b95dc0a3c05

Spamming screwdriver results in warns being written to server console every time:
![image](https://github.com/space-wizards/space-station-14/assets/47284450/fe2dd9f3-dec0-43f7-b659-119bf8fe9ca3)

**Comment by tetonis** [comment]
Small note: This actually applies to every single component of the Particle Accelerator (Power box, end cap, etc.)
They are all using the wrong graph so you can't add the LV Cables to finish construction

**Comment by tetonis** [comment]
Mistake

**Comment by arimah** [comment]
Hi! I did some investigating of this issue, trying desperately to figure out WHY it gets the wrong construction graph, and now the answer is as clear as day – or, a day of patchy clouds and occasional rain, at any rate.

In [ConstructionSystem.Graph.cs](https://github.com/space-wizards/space-station-14/blob/4d845caa0725d9a07f9f2bea1a5e85fc2009854d/Content.Server/Construction/ConstructionSystem.Graph.cs#L327-L331), we find this:

```csharp
// We set the graph and node accordingly.
ChangeGraph(newUid, userUid, construction.Graph, construction.Node, false, newConstruction);

if (construction.TargetNode is {} targetNode)
    SetPathfindingTarget(newUid, targetNode, newConstruction);
```

What this code does is to _overwrite_ `newConstruction` with the _old_ graph. As far as I can tell, it seems the PA emitter parts are the _only_ constructible objects that require a change in construction graph; everything else just has a single graph, with machines stopping at the `Machine` graph's `machine` node. This means:

1. `construction` is set to the completed state (e.g. `Machine/machine` for machines) because we have just completed construction.
2. For `newConstruction`, exactly one of the following is true:
    1. The new entity does not require a new construction graph: `newConstruction` is set to the completed state (e.g. `Machine/machine` for machines), matching `construction`, and the call to `ChangeGraph()` in practice just wastes time (with some `UpdatePathfinding()`).
    3. The new entity requires a different graph for additional completion steps: `newConstruction` is already set to the new graph with the final node (e.g. `completed`) as the target, and the call to `ChangeGraph()` overwrites it with the old graph, making it impossible to finish construction.

The following change makes the PA parts constructible, but I don't know if it has any other effects on the construction system:

```csharp
// Don't know if this call is even necessary...
UpdatePathfinding(newUid, newConstruction);

// If the new entity has the *same* construction graph, stay on the same node.
// If not, we effectively restart the construction graph, so the new entity can be completed.
if (construction.Graph == newConstruction.Graph)
{
    ChangeNode(newUid, userUid, construction.Node, false, newConstruction);

    // Retain the target node if an entity change happens in response to deconstruction.
    if (construction.TargetNode is {} targetNode)
    {
        SetPathfindingTarget(newUid, targetNode, newConstruction);
    }
}
```

Interestingly, it seems, both before and after the above change, an already fully constructed PA part cannot be deconstructed into its machine frame: you can't return to the `Machine` graph in any way. However, a broken PA part _can_ be deconstructed fully, because it's still using the `Machine` graph. What a delightful mess this whole construction system is!

TL;DR: I have figured out the root cause but don't know if my fix is valid, will probably open a PR to get code feedback.

---

I have now written the word "machine" so many times it has lost all meaning.
